pkg/evaluator/evaluator.go

```diff
@@ -367,6 +398,20 @@ func (e *Evaluator) evalFunccall(funcCall *parser.FuncCall) (value, error) {
 	builtin, ok := e.builtins.Funcs[funcCall.Name]
 	if ok {
 		val, err := builtin.Func(e.scope, args)
+		if funcCall.Name == "assert" {
```

Comment:
I would like to have a better way to do this, and it could be done by wrapping the assert builtin function if they were passed the `funcCall` value available here. It would be quite a bit of churn to update every builtin function to take a `*parser.FuncCall` argument as well (and have every existing one ignore it) so that's probably not quite right, but if the builtinFunc was an interface and not a function, it could be asserted to an interface with an expanded signature and have the call here pass that funcCall in that case, otherwise call the other way.

I think for now this is fine, but I might have a play and see if it is not too churny to do it this other way.

:pineapple: